assets/dns/daemonset.yaml

```diff
@@ -44,14 +44,44 @@ spec:
           timeoutSeconds: 5
           successThreshold: 1
           failureThreshold: 5
+      - name: dns-node-resolver
+        image: "openshift/origin-cli:v4.0"
+        imagePullPolicy: IfNotPresent
+        securityContext:
+          privileged: true
+        volumeMounts:
+        - name: hosts-file
+          mountPath: /etc/hosts
+        # env NAMESERVER is set at runtime
+        command:
+        - /bin/bash
+        - -c
+        - |
+          #!/bin/bash
+          set -euo pipefail
+
+          REGISTRY_SERVICE="docker-registry.default.svc.cluster.local"
+          OPENSHIFT_MARKER="openshift-generated-node-resolver"
+          HOSTS_FILE="/etc/hosts"
+          TEMP_FILE="/etc/hosts.tmp"
+
+          while true; do
+            ( grep -v "${OPENSHIFT_MARKER}" "${HOSTS_FILE}"; dig @"${NAMESERVER}" +short "${REGISTRY_SERVICE}" | sed 's/$/ '"${REGISTRY_SERVICE}"' # '"${OPENSHIFT_MARKER}"'/' ) > "${TEMP_FILE}" && \
+            cp "${TEMP_FILE}" "${HOSTS_FILE}"
+            sleep 60
+          done
```

Comment:
Another approach to consider which doesn't require `sed` or a subshell:

```bash
# Build a new hosts file from /etc/hosts with our custom entries filtered out
grep -v "${OPENSHIFT_MARKER}" "${HOSTS_FILE}" > "${TEMP_FILE}"

# Append refreshed custom entries to the new hosts file
dig @"${NAMESERVER}" +short "${REGISTRY_SERVICE}"  >> "${TEMP_FILE}"

# Replace /etc/hosts with our modified version
cp -f "${TEMP_FILE}" "${HOSTS_FILE}"
```

Reply:
Alternatively, replace parentheses with braces to group commands without forking a subshell.

Reply:
You do still need the host name in the appended hosts entry, but you can avoid using sed:
```bash
printf '%s %s\n' "$(dig "@$NAMESERVER" +short "$REGISTRY_SERVICE")" "$REGISTRY_SERVICE"
```
(`echo` would work too.)